src/utils/bind-params.js

```diff
@@ -26,4 +26,32 @@ const getBindParams = (body, allowedProperties, baseValue = {}) => {
   return bindParams;
 };
 
-export { getBindParams };
+/**
+ * convert an output bind param name like `nameOut` to a
+ * property name like `name` by removing 'Out' at the end of the string
+ *
+ * @param {string} outBindParamName
+ * @returns {string} the property name
+ */
+const outBindParamToPropertyName = (outBindParamName) => {
+  const outBindEndRegex = /(.*)Out$/;
+  const regexResults = outBindEndRegex.exec(outBindParamName);
+  return regexResults ? regexResults[1] : outBindParamName;
+};
+
+/**
+ * Converts the return value of a SQL query that uses
+ * `RETURNS ... INTO ...` into the format of the return
+ * from a `SELECT` query to it can be passed directly to
+ * `serializeIngredient`.
+ *
+ * @param {object} outBinds
+ * @returns {object}
+ */
+const convertOutBindsToRawResource = (outBinds) => _.reduce(outBinds,
+  (rawResource, bindValueArray, bindName) => {
+    [rawResource[outBindParamToPropertyName(bindName)]] = bindValueArray;
+    return rawResource;
+  }, {});
```

Comment:
If I'm understanding this right, you're taking an object `outBinds` and changing all of its key names using `outBindParamToPropertyName()`, right? If this is the case, I think [`_.mapKeys()`](https://lodash.com/docs/4.17.15#mapKeys) might be more helpful and simpler here.

Reply:
this is an excellent idea haha, I can't believe I didn't notice this

Reply:
Ah, now I understand why I used `reduce`. This:
```js
[rawResource[outBindParamToPropertyName(bindName)]]
```

Is actually destructing the single value array in `bindValueArray`, sort of like saying `rawResource[foo] = bindValueArray[0]`. So the fact that it's transforming keys and values means (I think) `reduce` is actually the best function to use here.

Reply:
I am open to suggestions if it seems like there's a better way to do this, I think the fact that both of us got confused about what this function actually does is not a vote of confidence in its quality.

Reply:
Ah I see, good catch. Lodash also has a `_.transform()` function which basically works like reduce but you don't need to explicitly return the new result, making it better when transforming an object like this. As for the array destructuring, yeah it's a little easy to miss I think. Two options I can think of are

1. Add an intermediate variable
    ```js
    const [bindValue] = bindValueArray
    rawResource[outBindParamToPropertyName(bindName)] = bindValue;
    ```
2. just add a comment about how you're taking the first value in the array for its new value.

It seems to me that the array destructuring is more noticeable for a single variable name rather than surrounding a larger statement. Object destructuring doesn't seem to have this problem because we put spaces around the `{}`. Any other thoughts?

Reply:
I think the first option is probably the best, since it lets us avoid using a comment. Adding in `transform` as well makes for a huge improvement:

```js
const convertOutBindsToRawResource = (outBinds) => _.transform(outBinds,
  (rawResource, bindValueArray, bindName) => {
    const [bindValue] = bindValueArray;
    rawResource[outBindParamToPropertyName(bindName)] = bindValue;
  }, {})
```

I'd agree that extra spacing makes object destructuring a little easier to read, although I think it's also just harder to use it in cases where it gets nested in an expression like this.